plugins/linux-v4l2/v4l2-output.c

```diff
@@ -125,19 +138,57 @@ static bool try_connect(void *data, const char *device)
 
 	vcam->device = open(device, O_RDWR);
 
-	if (vcam->device < 0)
+	if (vcam->device == -1) {
+		blog(LOG_ERROR, "Failed to open device '%s' (%d)", device,
+		     errno);
 		return false;
+	}
+
+	if (ioctl(vcam->device, VIDIOC_QUERYCAP, &capability) == -1) {
+		blog(LOG_WARNING,
+		     "Failed to query device capabilities on '%s' (%d)", device,
+		     errno);
+		goto fail_close_device;
+	}
 
-	if (ioctl(vcam->device, VIDIOC_QUERYCAP, &capability) < 0)
+	blog(LOG_DEBUG,
```

Comment:
Personally I rather guard debug output with a check for the `DEBUG` preprocessor definition so the code doesn't even make it into optimised builds.